pepys_import/core/store/common_db.py

```diff
@@ -189,13 +196,18 @@ def validate(
                 return True
             return False
         elif validation_level == validation_constants.ENHANCED_LEVEL:
-            for measurement in self.measurements[parser]:
-                BasicValidator(measurement, errors, parser)
-                for basic_validator in LOCAL_BASIC_VALIDATORS:
-                    basic_validator(measurement, errors, parser)
-                EnhancedValidator(measurement, errors, parser)
-                for enhanced_validator in LOCAL_ENHANCED_VALIDATORS:
-                    enhanced_validator(measurement, errors, parser)
+            for key, objects in self.measurements[parser].items():
+                for index, curr_object in enumerate(objects):
```

Comment:
@BarisSari - I don't think this algorithm can handle when a datafile holds data for multiple ships.  I'm pretty sure we need a dictionary of previous objects, indexed by either platform name or sensor.

Oh, and here's a simpler logic for remembering the previous value (though it will need modifying for the multiple platforms):
```
prev_object = None
for index, curr_object in enumerate(objects):
    if index >= 1:
    [...]
    for enhanced_validator in LOCAL_ENHANCED_VALIDATORS:
        enhanced_validator(curr_object, errors, parser, prev_object)
    prev_object = curr_object
```

Reply:
Hey, don't I look silly for not reading your PR description 🤣 

I presume your code handles just the problem I was discussing, right @BarisSari ?

Reply:
That's fine 😄 yes, I think we already have example datafiles that hold data for multiple ships. The algorithm handles them without any problem

Reply:
@BarisSari  - I think this is still simpler logic for remembering the previous measurement:
```
prev_object = None
for index, curr_object in enumerate(objects):
    [...]
    for enhanced_validator in LOCAL_ENHANCED_VALIDATORS:
        enhanced_validator(curr_object, errors, parser, prev_object)
    prev_object = curr_object
```

Reply:
We should keep the measurement objects of the datafile in any case. Because we are committing them to the DB when they are validated. I changed only the dictionary (self.measurements), so that we can reach the previous objects easier. So, we don't need to keep track of the previous objects in any other list or variable. Right now, we aren't bothered with the correct assignment of `prev_object` according to its' platform because we are already keeping them according to their platform. So, it can't be wrong. Don't you think that it's more convenient?

Reply:
I may be misinterpreting the code. Well, or not explaining myself.  We currently have this:
```
        elif validation_level == validation_constants.ENHANCED_LEVEL:
            for key, objects in self.measurements[parser].items():
                for index, curr_object in enumerate(objects):
                    prev_object = None
                    if index >= 1:
                        prev_object = objects[index - 1]
                   # 
                   # do tests
                   # 
```

I'm proposing it would be simpler like this:
```
        elif validation_level == validation_constants.ENHANCED_LEVEL:
            for key, objects in self.measurements[parser].items():
                prev_object = None
                for curr_object in enumerate(objects):
                   # 
                   # do tests
                   # 
                   prev_object = curr_object
```
In the latter version we're not using/checking the index counter, or doing array access.

Reply:
Okay, @IanMayo . I understood what you're proposing. I modified it a bit for keeping track of objects according to their platform. There is no index counter and array access anymore.

https://github.com/debrief/pepys-import/blob/c58b2fc3ef53f685ff9acb79d21a9996c2ce1e7c/pepys_import/core/store/common_db.py#L200-L212